gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,209 @@
+
+                        pair_scores[key_a_id, key_b_id] += score_a_b
+                        pair_scores[key_b_id, key_a_id] += score_a_b
+
+                        depth_collection_views[key_a_id].append(cam_a_pos_i[-1])
+                        depth_collection_views[key_b_id].append(cam_b_pos_i[-1])
+
+        depth_metas[:, 0] = np.array([np.floor(np.min(depth_collection_views[i])) for i in range(num_images)])
+        depth_metas[:, 1] = np.array([np.ceil(np.max(depth_collection_views[i])) for i in range(num_images)])
+        pairs = np.argsort(pair_scores, axis=0)[:, -self.num_views + 1 :][:, ::-1]
+
+        return pairs, depth_metas
+
+    def __len__(self) -> int:
+        """Get the number of images
+
+        Returns:
+            length of image dictionary's keys
+        """
+
+        return self.num_images
+
+    def __getitem__(self, index: int) -> Dict[str, Any]:
+        """Get one test input to Patchmatch Net, Wang's work in https://github.com/FangjinhuaWang/PatchmatchNet.git is
+            referred to in this method
+
+        Args:
+            index: index of yield item
+
+        Returns:
+            return dictionary contains:
+                "idx" test image index: int
+                "imgs" source and reference images: (num_views, image_channel, image_h, image_w)
+                "proj_matrices" projection matrices: (num_views, 4, 4)
+                "depth_min" minimum depth: int
+                "depth_max" maximum depth: int
+                "filename" output filename pattern: string
+        """
+        ref_key = self.keys[index]
+        src_keys = [self.keys[src_index] for src_index in self.pairs[index]]
+
+        cam_keys = [ref_key] + src_keys
+
+        imgs_0 = []
+        imgs_1 = []
+        imgs_2 = []
+        imgs_3 = []
+        proj_matrices_0 = []
+        proj_matrices_1 = []
+        proj_matrices_2 = []
+        proj_matrices_3 = []
+
+        for cam_key in cam_keys:
+            img = self.images[cam_key].value_array
+            np_img = np.array(img, dtype=np.float32) / 255.0
+            np_img = cv2.resize(np_img, (self.image_w, self.image_h), interpolation=cv2.INTER_LINEAR)
+
+            h, w, _ = np_img.shape
+
+            imgs_0.append(np_img)
+            imgs_1.append(cv2.resize(np_img, (w // 2, h // 2), interpolation=cv2.INTER_LINEAR))
+            imgs_2.append(cv2.resize(np_img, (w // 4, h // 4), interpolation=cv2.INTER_LINEAR))
+            imgs_3.append(cv2.resize(np_img, (w // 8, h // 8), interpolation=cv2.INTER_LINEAR))
+
+            # intrinsics, extrinsics = self.data["cameras"][vid]
+            intrinsics = self.sfm_result.get_camera(cam_key).calibration().K()
+            extrinsics = np.linalg.inv(self.sfm_result.get_camera(cam_key).pose().matrix())
+
+            # multiply intrinsics and extrinsics to get projection matrix
+            proj_mat = extrinsics.copy()
+            intrinsics[:2, :] *= 0.125
```

Comment:
prefer 1/8 instead of 0.125

Can this go in a loop, to prevent the copy-paste?